rcl/test/rcl/test_wait.cpp

```diff
@@ -269,15 +273,16 @@ TEST(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), multi_wait_set_threaded)
   //   ss << "]";
   //   printf("%s\n", ss.str().c_str());
   // };
+
   size_t loop_count = 0;
   while (loop_test()) {
     loop_count++;
-    // print_state("triggering, current states: ");
     for (const auto & test_set : test_sets) {
       ret = rcl_trigger_guard_condition(&test_set.guard_condition);
       EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
     }
-    std::this_thread::sleep_for(trigger_period);
+    std::this_thread::yield();
```

Comment:
Again, not sure why we need this when we're sleeping. Usually you use `yield` when doing a busy wait, like this:

```c++
while (!condition) {
  std::this_thread::yield();
}
```

So that your busy wait doesn't starve the other threads. I think `yield` is basically implied in other blocking system calls like sleep, select, and in our case probably the DDS wait function too.